libtenzir/builtins/aggregation-functions/distinct.cpp

```diff
@@ -90,7 +94,40 @@ class distinct_function final : public aggregation_function {
     distinct_ = {};
 };
 
-class plugin : public virtual aggregation_function_plugin {
+class distinct_instance final : public aggregation_instance {
+public:
+  explicit distinct_instance(ast::expression expr) : expr_{std::move(expr)} {
+  }
+
+  auto update(const table_slice& input, session ctx) -> void override {
+    auto arg = eval(expr_, input, ctx);
+    if (caf::holds_alternative<null_type>(arg.type)) {
+      return;
+    }
+    // XXX: What should happen if it receives different types?
+    // Currently, everything ends up coerced to strings.
+    for (auto i = int64_t{}; i < arg.array->length(); ++i) {
+      if (arg.array->IsNull(i)) {
+        continue;
+      }
+      distinct_.insert(materialize(value_at(arg.type, *arg.array, i)));
+    }
+  }
+
+  auto finish() -> data override {
+    if (distinct_.size() == 0) {
+      return data{};
+    }
+    return list{distinct_.begin(), distinct_.end()};
+  }
```

Comment:
Here, you say that `distinct` over an empty sequence should return `null`. Why not the empty list?

Reply:
True, changed. Thanks